src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunCodegenCompilation.cs

```diff
@@ -448,7 +448,7 @@ private void RewriteComponentFile(string inputFile, string outputFile, string ow
                 ReadyToRunFlags.READYTORUN_FLAG_Component |
                 ReadyToRunFlags.READYTORUN_FLAG_NonSharedPInvokeStubs;
 
-            if (inputModule.IsPlatformNeutral)
+            if (ReadyToRunCodegenCompilationBuilder.IsReadyToRunPlatformNeutral(inputModule.PEReader) || inputModule.IsPlatformNeutral)
```

Comment:
```suggestion
            if (inputModule.IsPlatformNeutral || ReadyToRunCodegenCompilationBuilder.IsReadyToRunPlatformNeutralSource(inputModule.PEReader))
```
Nit: Check the common case first